### PR TITLE
Convert to markdown onnx rst

### LINKED ISSUE #155030: Convert to markdown: onnx_dynamo_onnxruntime_backend.rst, onnx_dynamo.rst, onnx_ops.rst, onnx_torchscript_supported_aten_ops.rst, onnx_torchscript.rst
Convert the following files to MyST markdown and ensure that the doc tests are green:

- [onnx_dynamo_onnxruntime_backend.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_dynamo_onnxruntime_backend.rst)
- [onnx_dynamo.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_dynamo.rst)
- [onnx_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_ops.rst)
- [onnx_torchscript_supported_aten_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript_supported_aten_ops.rst)
- [onnx_torchscript.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript.rst)

To convert an `.rst` file:

1) Assign the issue to yourself using the `/assigntome` comment.

2) **Important!** Use `git mv filename.rst filename.md` to convert the issue from `.rst` to `.md`. **DO NOT delete**
   the `.rst` and create the `.md` - we won't accept PRs with deleted `.rst` files. We want to preserve the git log of
   those files. Therefore, using `git mv` is very important!

3) Before submitting a PR, build the docs locally as described in the [README](https://github.com/pytorch/pytorch/blob/main/README.md#building-the-documentation).
   Preview localy with `make serve` to ensure that the HTML renders correctly.    

4) Run `lintrunner` from the root of the repository:

   ```
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

4) Submit a PR with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:

   * pull / linux-docs / build-docs-python-false (pull_request)
   * linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)

   Doc tests start after the Docker build, which might take about 30 minutes. 
   The doc build itself should take about 25 minutes. When the build is done, 
   you can see the preview of your build by clicking:
   📄 **Preview Python docs built from this PR** in the `pytorch-bot` comment.

   Make sure your PR is tagged with `docathon-h1-2025`, `module: docs`, `topic: not user facing`,
   and `topic: docs` labels.

5) Compare the preview of the pages you converted with same pages it the main branch.
   The converted pages in preview must look the same as in the main branch.

6) Once the PR is accepted, one of the maintainers will merge it.

See [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown)
in our wiki or [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html) for detailed syntax.

Here is an example [PR](https://github.com/pytorch/pytorch/pull/154438)

cc @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by nirajkamal**
Fixes #155030

Converts the following files to MyST markdown and ensure that the doc tests are green:

- [x] [onnx_dynamo_onnxruntime_backend.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_dynamo_onnxruntime_backend.rst)
- [x] [onnx_dynamo.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_dynamo.rst)
- [x] [onnx_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_ops.rst)
- [onnx_torchscript_supported_aten_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript_supported_aten_ops.rst) - not changed as it is autogenerated
- [onnx_torchscript.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript.rst) - fixed in #155390

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

**Comment by nirajkamal** [comment]
I am using  git mv filename.rst filename.md  - but want to make sure if this is correct and if the git logs are preserved, thus this draft PR.

cc - @svekars @spzala

**Comment by spzala** [comment]
> I am using git mv filename.rst filename.md - but want to make sure if this is correct and if the git logs are preserved, thus this draft PR.
> 
> cc - @svekars @spzala

Thanks @nirajkamal !! Yes, the renaming looks good to me (first commit). However, the changes shows as .rst deleted and .md added, so let's wait for feedback from @svekars

**Comment by svekars** [comment]
This approach seems to be okay, please feel feel free to work on other files.

**Comment by nirajkamal** [comment]
@svekars in  [onnx_torchscript_supported_aten_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript_supported_aten_ops.rst)

it says
```
.. This file is automatically generated during the documentation build
.. by cross referencing ONNX operator symbolics with TorchScript operators via
.. ``docs/source/scripts/build_onnx_torchscript_supported_aten_op_csv_table.py``.
.. Do not modify directly and instead `rebuild the docs <https://github.com/pytorch/pytorch#building-the-documentation>`_.
```

should I modify it!?

**Comment by nirajkamal** [comment]
Other files are modified

**Comment by justinchuby** [comment]
> @svekars in [onnx_torchscript_supported_aten_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript_supported_aten_ops.rst)
> 
> it says
> 
> ```
> .. This file is automatically generated during the documentation build
> .. by cross referencing ONNX operator symbolics with TorchScript operators via
> .. ``docs/source/scripts/build_onnx_torchscript_supported_aten_op_csv_table.py``.
> .. Do not modify directly and instead `rebuild the docs <https://github.com/pytorch/pytorch#building-the-documentation>`_.
> ```
> 
> should I modify it!?

Feel free to remove both files if that makes it easier, thanks

**Comment by nirajkamal** [comment]
> > @svekars in [onnx_torchscript_supported_aten_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript_supported_aten_ops.rst)
> > it says
> > ```
> > .. This file is automatically generated during the documentation build
> > .. by cross referencing ONNX operator symbolics with TorchScript operators via
> > .. ``docs/source/scripts/build_onnx_torchscript_supported_aten_op_csv_table.py``.
> > .. Do not modify directly and instead `rebuild the docs <https://github.com/pytorch/pytorch#building-the-documentation>`_.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > should I modify it!?
> 
> Feel free to remove both files if that makes it easier, thanks

@justinchuby I was thinking rather not to modify [onnx_torchscript_supported_aten_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript_supported_aten_ops.rst) as it is an autogenerated file

Are you suggesting to remove [onnx_torchscript_supported_aten_ops.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript_supported_aten_ops.rst) and docs/source/scripts/build_onnx_torchscript_supported_aten_op_csv_table.py ?

**Comment by justinchuby** [comment]
Keeping it is ok to be on the safe side, thanks! The page was a little outdate and we may remove it at some point.

**Comment by svekars** [comment]
yes, please leave that one as is.

**Comment by nirajkamal** [comment]
There are no issues in lint runner in my local. Making this ready for review

**Comment by nirajkamal** [comment]
pr sanity check fails due to the number of lines changed
`Your PR is 2037 LOC which is more than the 2000 maximum
allowed within PyTorch infra. PLease make sure to split up
your PR into smaller pieces that can be reviewed.
If you think that this rule should not apply to your PR,
please contact @albanD or @seemethere.`

File: docs/source/onnx_ops.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    (input_data, cos_cache_data, sin_cache_data, position_ids_data),
+    dynamic_shapes=dynamic_shapes,
+    dynamo=True,
+    opset_version=23,
+)
+```
+
+Printing the ONNX program will show the ONNX operators used in the graph:
+
+```{code-block} python
```
Comment (svekars) [inline docs/source/onnx_ops.md]:
```suggestion
```python
```

**Comment by svekars** [comment]
Can you move one of the files into a separate PR?

File: docs/source/onnx_ops.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    (input_data, cos_cache_data, sin_cache_data, position_ids_data),
+    dynamic_shapes=dynamic_shapes,
+    dynamo=True,
+    opset_version=23,
+)
+```
+
+Printing the ONNX program will show the ONNX operators used in the graph:
+
+```python
```
Comment (svekars) [inline docs/source/onnx_ops.md]:
```suggestion
```
```

File: docs/source/onnx_ops.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    model,
+    (input_data, cos_cache_data, sin_cache_data, position_ids_data),
+    dynamic_shapes=dynamic_shapes,
+    dynamo=True,
+    opset_version=23,
+)
+```
+
+Printing the ONNX program will show the ONNX operators used in the graph:
+
```
Comment (svekars) [inline docs/source/onnx_ops.md:62]:
```suggestion
```
```

**Comment by nirajkamal** [comment]
> Can you move one of the files into a separate PR?

@svekars shall I do `git checkout [commit ID] -- path/to/file.rst` such that logs are preserved? Is there any other recommended way such that other recent changes are preserved? I am thinking of taking [onnx_torchscript.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript.rst) to a new PR.

File: docs/source/onnx_ops.md (hunk 2 above)
Comment (nirajkamal) [reply docs/source/onnx_ops.md]: I have done ```` ```{code-block} <language name>```` like in MyST docs.
Shall I change it to `` ```<language name>`` instead for all code blocks

**Comment by svekars** [comment]
> shall I do `git checkout [commit ID] -- path/to/file.rst` such that logs are preserved? Is there any other recommended way such that other recent changes are preserved? I am thinking of taking [onnx_torchscript.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_torchscript.rst) to a new PR.

I've found that `git mv` works pretty well for this.

File: docs/source/onnx_dynamo.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+assert onnx_program.model.graph.inputs[0].shape == ("batch_size", 8)
+assert onnx_program.model.graph.inputs[1].shape == ("batch_size", 8)
+assert onnx_program.model.graph.inputs[2].shape == ("batch_size", 8)
+```
+
+As the code above shows, all you need is to provide {func}`torch.onnx.export` with an instance of the model and its input.
+The exporter will then return an instance of {class}`torch.onnx.ONNXProgram` that contains the exported ONNX graph along with extra information.
+
+The in-memory model available through ``onnx_program.model_proto`` is an ``onnx.ModelProto`` object in compliance with the [ONNX IR spec](https://github.com/onnx/onnx/blob/main/docs/IR.md).
+The ONNX model may then be serialized into a [Protobuf file](https://protobuf.dev/)_ using the :meth:`torch.onnx.ONNXProgram.save` API.
```
Comment (svekars) [inline docs/source/onnx_dynamo.md]:
```suggestion
The ONNX model may then be serialized into a [Protobuf file](https://protobuf.dev/) using the {meth}`torch.onnx.ONNXProgram.save` API.
```

File: docs/source/onnx_ops.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+.. autofunction:: torch.onnx.ops.symbolic_multi_out
+```
+
+
+## ONNX Operators
+
+The following operators are implemented as native PyTorch ops and can be exported as
+ONNX operators. They can be used natively in an ``nn.Module``.
+
+For example, you can define a module::
```
Comment (svekars) [inline docs/source/onnx_ops.md]:
```suggestion
For example, you can define a module:
```

**Review by svekars — COMMENTED** [review]
Minor formatting fixes

File: docs/source/onnx_dynamo.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+
+## Dependencies
+
+The ONNX exporter depends on extra Python packages:
+
+  - [ONNX](https://onnx.ai)
+  - [ONNX Script](https://microsoft.github.io/onnxscript)
+
+They can be installed through [pip]{https://pypi.org/project/pip/):
```
Comment (svekars) [inline docs/source/onnx_dynamo.md]:
```suggestion
They can be installed through [pip](https://pypi.org/project/pip/):
```

**Review by svekars — APPROVED** [review]
